src/App.tsx

```diff
@@ -1,26 +1,177 @@
+        setError('Unable to delete a todo');
+      })
+      .finally(() => {
+        setTempIds(currentIds =>
+          currentIds.filter(deletedId => idNumber !== deletedId),
+        );
+      });
+  };
+
+  const deleteOnlyCompleted = () => {
+    const removeDeletedTodos = todos.filter(todo => !todo.completed);
+
+    setTodos(removeDeletedTodos);
+  };
+
+  const filteredTodos = (
+    filtrTodos: Todo[],
+    filterStatus: FilterButtons,
+  ): Todo[] => {
+    const updateTodos = [...filtrTodos];
+
+    if (filterStatus) {
+      switch (filterStatus) {
+        case FilterButtons.Active:
+          return updateTodos.filter(todo => !todo.completed);
+        case FilterButtons.Completed:
+          return updateTodos.filter(todo => todo.completed);
+        default:
+          break;
+      }
+    }
+
+    return updateTodos;
+  };
+
+  const allTodosAreCompleted =
+    todos.filter(todo => todo.completed).length === todos.length;
+
+  const handleChangingStatusButton = () => {
+    if (allTodosAreCompleted) {
+      const changeTodos: Todo[] = todos.map(todo => {
+        return {
+          ...todo,
+          completed: false,
+        };
+      });
+
+      setTodos(changeTodos);
+    } else {
+      const changeTodo = todos.map(todo => {
+        return {
+          ...todo,
+          completed: true,
+        };
+      });
+
+      setTodos(changeTodo);
+    }
+  };
+
+  const onlyActiveTodos = todos.filter(todo => !todo.completed);
+
+  const todosCounter =
+    todos.length === 1
+      ? 'one item left'
+      : `${onlyActiveTodos.length} items left`;
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        {' '}
+        <TodoHeader
+          addTodos={handleAddingTodos}
+          toggleButton={handleChangingStatusButton}
+          todos={todos}
+          allTodosAreCompleted={allTodosAreCompleted}
+          toDoTitle={newToDoTitle}
+          setToDoTitle={setNewToDoTitle}
+          error={error}
+        />
+        <TodoList
+          todos={todos}
+          filteredTodos={filteredTodos}
+          filter={filterButton}
+          deleteTodo={handleDeletedTodo}
+          loadingTodos={tempIds}
+          temporaryTodo={temporaryTodo}
+        />
+        <TodoFooter
+          todos={todos}
+          filter={filterButton}
+          setFilter={setFilterButton}
+          deleteAllCompleted={deleteOnlyCompleted}
+          todosCounter={todosCounter}
+        />
+      </div>
+
+      <Error error={error} setError={setError} />
```

Comment:
create handler for setError instead of passing it as props [[more info]](https://blog.stackademic.com/dont-pass-setstate-as-a-prop-2cc2b187d323)